courses-rest/src/main/java/no/sikt/nva/fs/client/HttpUrlConnectionFsClient.java

```diff
@@ -0,0 +1,89 @@
+package no.sikt.nva.fs.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.PasswordAuthentication;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.Base64;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpUrlConnectionFsClient implements FsClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpUrlConnectionFsClient.class);
+    public static final  String UNEXPECTED_STATUS_CODE_LOG_MESSAGE_PREFIX = "Unexpected response code from FS: ";
+    private static final String TEACHING_URI_PATH = "undervisning";
+    private static final String DB_IDENTIFIER_QUERY_PARAM_NAME = "dbId";
+    private static final String INSTITUTION_QUERY_PARAM_NAME = "emne.institusjon";
+    private static final String YEAR_QUERY_PARAM_NAME = "semester.ar";
+    private static final String LIMIT_QUERY_PARAM_NAME = "limit";
+    private static final String UNLIMITED = "0";
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper;
+    private final String baseUri;
+    private final int institutionCode;
+    private final String username;
+    private final String password;
+
+    public HttpUrlConnectionFsClient(final ObjectMapper objectMapper, final String baseUri,
+                                     final int institutionCode, final String username,
+                                     final String password) {
+
+        this.httpClient = HttpClient.newBuilder()
+                              .authenticator(new Authenticator() {
+                                  @Override
+                                  protected PasswordAuthentication getPasswordAuthentication() {
+                                      return new PasswordAuthentication(username, password.toCharArray());
+                                  }
+                              })
```

Comment:
Må man både sette opp authenticator og sette headerne i requesten? holder det ikke med bare å sette i header? Spør fordi jeg ikke vet. Ikke fordi det må fikses

Reply:
Du har rett. Fjernet manuell setting av header, og da fikk jeg følgende problem: https://stackoverflow.com/questions/54208945/java-11-httpclient-not-sending-basic-authentication

Jeg har fjernet Authenticator og setter istedenfor header manuelt.